content/en/guide/v10/components.md

```diff
@@ -94,7 +94,9 @@ In order to have the clock's time update every second, we need to know when `<Cl
 | `getSnapshotBeforeUpdate(prevProps, prevState)` | called just before `render()`. return value is passed to `componentDidUpdate`.
 | `componentDidUpdate(prevProps, prevState, snapshot)` | after `render()`
 
-> See [this diagram](https://twitter.com/dan_abramov/status/981712092611989509) to get a visual overview of how they relate to each other.
+Here's a visual overview of how they relate to each other (originally posted in [a tweet](https://web.archive.org/web/20191118010106/https://twitter.com/dan_abramov/status/981712092611989509) by Dan Abramov):
+
+![Diagram of lifecycle methods, depicting mounting, updating, and unmounting on the x-axis, and render phase, pre-commit phase, and commit phase on the y-axis](https://github.com/tinymachine/preact-www/assets/171986/ea5fd5d1-0b9e-4125-9ca0-b94d544c0318)
```

Comment:
That's a hotlink, which we'd need to avoid. We'd want the image uploaded to this repo, in the [assets directory](https://github.com/preactjs/preact-www/tree/master/src/assets). Here's an example of how to [reference the image path](https://github.com/preactjs/preact-www/blob/e06b7f9eb9e0958f9b17c4a7efd8dbe14c72c32e/content/en/blog/introducing-signals.md?plain=1#L56) from markdown.

Exactly as you say, our docs would now depend on a) you keeping your fork around forever and b) GitHub being willing to host our image for our site (essentially paying for our bandwidth). Both are risky.

While I get and appreciate the attempt at alt text, it's better left as "Diagram of lifecycle methods" (unless you really wanted to go all-out in a description -- certainly don't feel a need to). Describing the axis, without describing the content itself, doesn't present much value. That tells them how to read the graph, but not what data is actually shown.

Reply:
Thanks for the excellent feedback, @rschristian. I'll try to make those fixes tomorrow.

Reply:
No rush!

Thank you for not only opening an issue but also taking the time to correct it, we do really appreciate it.